From 473704933aa4867f637ce76110461c2ca811c331 Mon Sep 17 00:00:00 2001
From: Fang Hui <b31070@freescale.com>
Date: Fri, 28 Mar 2014 13:07:53 +0800
Subject: [PATCH 507/635] ENGR00305825 - [evk6sl]: camera apk, exception when
 STREAM_ON failed

mPreviewing should set true only after STREAM_ON success.

Signed-off-by: Fang Hui <b31070@freescale.com>
---
 mx6/libcamera/DeviceAdapter.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)
 mode change 100644 => 100755 mx6/libcamera/DeviceAdapter.cpp

diff --git a/mx6/libcamera/DeviceAdapter.cpp b/mx6/libcamera/DeviceAdapter.cpp
old mode 100644
new mode 100755
index 25c46a3..092e632
--- a/mx6/libcamera/DeviceAdapter.cpp
+++ b/mx6/libcamera/DeviceAdapter.cpp
@@ -1,6 +1,6 @@
 /*
  * Copyright (C) 2008 The Android Open Source Project
- * Copyright (C) 2012-2013 Freescale Semiconductor, Inc.
+ * Copyright (C) 2012-2014 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -409,7 +409,9 @@ status_t DeviceAdapter::startPreview()
     Mutex::Autolock lock(mPreviewBufsLock);
     ret = startDeviceLocked();
 
-    mPreviewing = true;
+    if(ret == 0) {
+        mPreviewing = true;
+    }
 
     return ret;
 }
-- 
1.8.0

